packages/mongodb-access-adapter/src/main/kotlin/com/mongodb/jbplugin/accessadapter/slice/BuildInfo.kt

```diff
@@ -14,19 +16,96 @@ import org.bson.Document
  * @see com.mongodb.jbplugin.accessadapter.MongoDbReadModelProvider.slice
  * @property version
  * @property gitVersion
+ * @property modules
+ * @property isLocalhost
+ * @property isDataLake
+ * @property isEnterprise
+ * @property isAtlas
+ * @property isLocalAtlas
+ * @property isAtlasStream
+ * @property isDigitalOcean
+ * @property isGenuineMongoDb
+ * @property nonGenuineVariant
+ * @property serverUrl
+ * @property buildEnvironment
  */
 data class BuildInfo(
     val version: String,
-    val gitVersion: String
+    val gitVersion: String?,
+    val modules: List<String>?,
+    val isLocalhost: Boolean,
+    val isDataLake: Boolean,
+    val isEnterprise: Boolean,
+    val isAtlas: Boolean,
+    val isLocalAtlas: Boolean,
+    val isAtlasStream: Boolean,
+    val isDigitalOcean: Boolean,
+    val isGenuineMongoDb: Boolean,
+    val nonGenuineVariant: String?,
+    val serverUrl: String,
+    val buildEnvironment: Map<String, String>,
 ) {
     object Slice : com.mongodb.jbplugin.accessadapter.Slice<BuildInfo> {
-        override suspend fun queryUsingDriver(from: MongoDbDriver): BuildInfo = from.runCommand(
-            Document(
-                mapOf(
-                    "buildInfo" to 1,
+        private val atlasRegex = Regex(""".*\.mongodb(-dev|-qa|-stage)?\.net$""")
+        private val atlasStreamRegex = Regex("""^atlas-stream-.+""")
+        private val isLocalhostRegex =
+            Regex(
+                "^(localhost" +
+                    "|127.([01]?[0-9][0-9]?|2[0-4][0-9]|25[0-5])" +
+                    ".([01]?[0-9][0-9]?|2[0-4][0-9]|25[0-5])" +
+                    ".([01]?[0-9][0-9]?|2[0-4][0-9]|25[0-5])" +
+                    "|0.0.0.0" +
+                    "|(?:0*:)*?:?0*1" +
```

Comment:
The IPv6 check here needs to account for [brackets] around the IP, right?

Reply:
It seems that the regex works differently in JS and Kotlin here. The [::1] IP works in mongodb-build-info, but had to fix it here 🤷